src/openms/source/FILTERING/DATAREDUCTION/IsotopeDistributionCache.cpp

```diff
@@ -37,92 +37,114 @@
 #include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/CoarseIsotopePatternGenerator.h>
 #include <OpenMS/DATASTRUCTURES/String.h>
 
+
 namespace OpenMS
 {
 
-  IsotopeDistributionCache::IsotopeDistributionCache(double max_mass, double mass_window_width, double intensity_percentage, double intensity_percentage_optional) :
-    mass_window_width_(mass_window_width)
+  IsotopeDistributionCache::IsotopeDistributionCache():
+    mass_window_width_(20.0),
+    intensity_percentage_(0.0),
+    intensity_percentage_optional_(0.0)
   {
-    Size num_isotopes = std::ceil(max_mass / mass_window_width) + 1;
+  }
 
-    //reserve enough space
-    isotope_distributions_.resize(num_isotopes);
 
+  void IsotopeDistributionCache::precalculateDistributionCache(Size num_begin, Size index)
+  {
+    isotope_distributions_.resize(index + 1);
+    distribution_cache_.resize(index + 1);
     //calculate distribution if necessary
-    for (Size index = 0; index < num_isotopes; ++index)
+    for (Size index = num_begin; index < isotope_distributions_.size() ; ++index)
     {
       //log_ << "Calculating iso dist for mass: " << 0.5*mass_window_width_ + index * mass_window_width_ << std::endl;
       CoarseIsotopePatternGenerator solver(20);
-      auto d = solver.estimateFromPeptideWeight(0.5 * mass_window_width + index * mass_window_width);
+      auto d = solver.estimateFromPeptideWeight(0.5*mass_window_width_ + index * mass_window_width_);
+
+      distribution_cache_[index] = d;
```

Comment:
we are storing a copy of `d` in the cache, and then go on to trim `d` and compute `isotope_distributions_[index]` based on it. Is this intended? If so, please document this somewhere